### PR TITLE
Fixed robots.txt 404 error

### LINKED ISSUE #2210: robots.txt 404 should be different
- Gitea version (or commit ref): 1.1.2 built with: bindata, sqlite
- Operating system: Ubuntu 16.04.2 LTS
- Database (use `[x]`):
  - [x] MySQL
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
  - [x] Site is down

## Description

If there is no custom robots.txt, it will return an empty 404 instead of the default gitea 404 error.

### PR DESCRIPTION
**Pull request by sinnlosername**
Fixed: https://github.com/go-gitea/gitea/issues/2210

### CODE REVIEW

File: routers/routes/routes.go [hunk 1]
```diff
@@ -651,7 +651,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 		if setting.HasRobotsTxt {
 			ctx.ServeFileContent(path.Join(setting.CustomPath, "robots.txt"))
 		} else {
-			ctx.Error(404)
+			ctx.Status(404)
```
Comment (lafriks) [inline routers/routes/routes.go]: It is not fixing the problem, correct code would be `ctx.Handle(404, "", nil)`

**Review by lafriks — CHANGES_REQUESTED** [review]

File: routers/routes/routes.go (hunk 1 above)
Comment (sinnlosername) [reply routers/routes/routes.go]: Sure? As I looked at other locations in the code where it returns a working 404.

Comment (lafriks) [reply routers/routes/routes.go]: I tried your code and it still returned empty 404 page

Comment (sinnlosername) [reply routers/routes/routes.go]: Okay, changed it!

Comment (lafriks) [reply routers/routes/routes.go]: `ctx.Status(404)` is used only in API routes

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Pull request merged**